### PR TITLE
lxd: Fix error message when deleting storage pools

### LINKED ISSUE #6697: Trying to delete a storage pool with instances from non-default projects gives "Error: No such object" error
Rather than the error `Error: storage pool "<pool>" has volumes attached to it` which is given when the pool contains instance volumes from the default project.

### PR DESCRIPTION
**Pull request by monstermunchkin**
This checks all projects for attached volumes when attempting to
delete a storage pool.

This fixes #6697.

Signed-off-by: Thomas Hipp <thomas.hipp@canonical.com>

### CODE REVIEW

File: lxd/storage_pools.go [hunk 1]
```diff
@@ -674,27 +674,41 @@ func storagePoolDeleteCheckPreconditions(cluster *db.Cluster, poolName string, p
...
+			if err != nil {
+				return response.InternalError(err)
+			}
 
-		for _, volume := range volumes {
-			if volume.Type != "image" {
-				return response.BadRequest(fmt.Errorf("storage pool \"%s\" has volumes attached to it", poolName))
+			for _, volume := range volumes {
+				if volume.Type != "image" {
+					return response.BadRequest(fmt.Errorf("storage pool \"%s\" has volumes attached to it", poolName))
```
Comment (tomponline) [inline lxd/storage_pools.go]: @monstermunchkin you could use `%q` here rather than `\"%s\"`

File: lxd/storage_pools.go [hunk 2]
```diff
@@ -674,27 +674,41 @@ func storagePoolDeleteCheckPreconditions(cluster *db.Cluster, poolName string, p
...
+		// Check if the storage pool is still referenced in any profiles.
+		profiles, err := profilesUsingPoolGetNames(cluster, project, poolName)
+		if err != nil {
+			return response.SmartError(err)
+		}
 
-	if len(profiles) > 0 {
-		return response.BadRequest(fmt.Errorf("Storage pool \"%s\" has profiles using it:\n%s", poolName, strings.Join(profiles, "\n")))
+		if len(profiles) > 0 {
+			return response.BadRequest(fmt.Errorf("Storage pool \"%s\" has profiles using it:\n%s", poolName, strings.Join(profiles, "\n")))
```
Comment (tomponline) [inline lxd/storage_pools.go]: @monstermunchkin you could use %q here rather than \"%s\"

File: lxd/storage_pools.go (hunk 1 above)
Comment (monstermunchkin) [reply lxd/storage_pools.go]: Fixed

File: lxd/storage_pools.go (hunk 2 above)
Comment (monstermunchkin) [reply lxd/storage_pools.go]: Fixed

**Pull request merged**